recorded_future/komand_recorded_future/util/demo_test.py

```diff
@@ -1,22 +1,55 @@
 import requests
+from komand.exceptions import ConnectionTestException
 
 
-def demo_test(token, logger):
+def demo_test(token):
     demo_url = "https://api.recordedfuture.com/v2/hash/demoevents?limit=1"
+    if not token:
+        raise ConnectionTestException(
+            cause="Missing API Key.",
+            assistance="Please provide API key."
+        )
     try:
         test_headers = {"X-RFToken": token}
         response = requests.get(demo_url, headers=test_headers)
-        response.raise_for_status()
-
-        return {"status": "200 OK"}
-
-    except requests.exceptions.HTTPError as e:
-        logger.error("HTTP error occurred. Error: " + str(e))
+        if 200 <= response.status_code < 300:
+            return {"status": "Success"}
+        elif response.status_code == 401:
+            raise ConnectionTestException(
+                cause="Invalid API Key.",
+                assistance="Please check that provided API key is correct."
+            )
+        elif response.status_code == 403:
+            raise ConnectionTestException(
+                cause="The account configured in your plugin connection is unauthorized to access this service.",
+                assistance="Verify the permissions for your account and try again."
+            )
+        elif response.status_code == 404:
+            raise ConnectionTestException(
+                cause="Invalid or unreachable endpoint provided.",
+                assistance="Verify the endpoint/URL/hostname configured in your plugin connection is correct."
+            )
+        elif response.status_code >= 400:
+            raise ConnectionTestException(
+                 cause="Something unexpected occurred.",
+                 assistance="Check the logs and if the issue persists please contact support.",
+                 data=response.text
+            )
     except requests.exceptions.ConnectionError as e:
-        logger.error("A network problem occurred. Error: " + str(e))
+        raise ConnectionTestException(
+            cause="A network problem occurred.",
+            assistance="Please try again and if the issue persists please contact support.",
+            data=e
+        )
     except requests.exceptions.Timeout as e:
-        logger.error("Timeout occurred. Error: " + str(e))
+        raise ConnectionTestException(
+            cause="Timeout occurred.",
+            assistance="Please try again and if the issue persists please contact support.",
+            data=e
+        )
     except requests.exceptions.TooManyRedirects as e:
-        logger.error("Too many redirects! Error: " + str(e))
-    except Exception as e:
-        logger.error("Error: " + str(e))
+        raise ConnectionTestException(
+            cause="Too many redirects.",
+            assistance="Please try again and if the issue persists please contact support.",
+            data=e
+        )
```

Comment:
Hi @dsliwinski-r7 - we have a number of preset ConnectionTestExceptions built into our SDK/runtime. Can you use those where possible? They can be found here https://github.com/rapid7/komand-plugin-sdk-python/blob/master/insightconnect_plugin_runtime/exceptions.py#L45

Reply:
Hi @mrinehart-r7 Sure, I can use them.